packages/app-builder-lib/src/targets/pkg.ts

```diff
@@ -118,9 +134,29 @@ export class PkgTarget extends Target {
     return { packagePath, packages }
   }
 
-  private async customizeDistributionConfiguration(distInfoFile: string, appPath: string, extraPackages: ExtraPackages | null) {
+  private async customizeDistributionConfiguration(distInfoFile: string, appPath: string, extraPackages: ExtraPackages | null, arch: Arch) {
     const options = this.options
-    const args = ["--synthesize", "--component", appPath]
+
+    // Build requirements plist for productbuild to generate correct hostArchitectures and allowed-os-versions
+    // This is the Apple-recommended way to specify architecture and OS requirements
+    // See: man productbuild, section "PRE-INSTALL REQUIREMENTS PROPERTY LIST"
+    const requirements: Record<string, any> = {}
+
+    // Set architecture based on build target - productbuild will generate correct hostArchitectures in distribution XML
+    // On macOS Big Sur+, productbuild defaults to both arm64 and x86_64 unless we specify otherwise
+    requirements.arch = archToAppleArchitectures(arch)
+
+    // Set minimum OS version - productbuild will generate allowed-os-versions in distribution XML
+    const minimumSystemVersion = this.packager.platformSpecificBuildOptions.minimumSystemVersion
+    if (minimumSystemVersion != null) {
+      requirements.os = [minimumSystemVersion]
+    }
+
+    // Write requirements plist to temp file
+    const requirementsPlistFile = path.join(this.outDir, "productbuild-requirements.plist")
```

Comment:
If possible, I'd like to use the temp file management electron-builder already leverages
```
const requirementsPlistFile = await this.packager.info.tempDirManager.getTempFile({ suffix: ".plist", prefix: "requirements" })
```

Reply:
Thanks Mike! Made requested change and pulled in latest.